### PR TITLE
TOK-403: Document rewards calculations for builders and backers

### PR DESCRIPTION
**Pull request by nmagariang**
## What

- Adding some documentation to explain how rewards are calculated throughout the cycles.

## Refs

- [Jira Ticket](https://rsklabs.atlassian.net/browse/TOK-403)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -359,23 +359,150 @@ Generate coverage report:
...
-TODO:
+The RoostockCollective Foundation requires builders to got through a KYC process and is in charge of submitting said
+approval as well as revoking it if necessary.
 
 ### Foundation treasury
 
-TODO:
+The RoostockCollective Foundation is in charge of holding and distributing the rewards from the program.
+
+## Rewards Distribution calculation
```
Comment (nmagariang) [inline README.md:382]: Readme might not be the right place for this

File: docs/RewardsCalculation.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Reward shares get updated every time there is a change in allocations (either adding or removing votes) based on the
+time left to end the cycle and after a distribution:
+
+- In case there is an increase in votes, the allocation deviation \* `timeUntilNextCycle` is added to the total rewards
+  shares
+- If votes are removed, then allocation deviation \* `timeUntilNextCycle` is removed from the total reward shares.
+- After a distribution, reward shares get reset to the total amount of allocation that the gauge has at the moment
+  multiplied by the duration of the cycle.
+
+The `backersManager` keeps track of the total amount of shares of every `Gauge` combined in `totalPotentialReward`.
```
Comment (antomor) [inline docs/RewardsCalculation.md:24]:
```suggestion
The `BackersManager` keeps track of the total amount of shares of every `Gauge` combined in `totalPotentialReward`.
```

File: docs/RewardsCalculation.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+On the other hand, the `backersManager` also keeps track of the `rewardsERC20` and `rewardsCoinbase`. They get updated
+each time the `backersManager` receives rewards through `notifyRewardAmount` and after each distribution, when they get
```
Comment (antomor) [inline docs/RewardsCalculation.md:27]:
```suggestion
each time the `BackersManager` receives rewards through `notifyRewardAmount` and after each distribution, when they get
```

File: docs/RewardsCalculation.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+(rewardShares * rewardsERC20) / totalPotentialReward
+```
+
+Coinbase rewards:
+
+```text
+ (rewardShares * rewardsCoinbase) / totalPotentialReward
+```
+
+The `backersManager` also keeps track of the `backerRewardPercentage`. This is passed to each gauge alongside the
```
Comment (antomor) [inline docs/RewardsCalculation.md:44]:
```suggestion
The `BackersManager` also keeps track of the `backerRewardPercentage`. This is passed to each gauge alongside the
```

File: docs/RewardsCalculation.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+of this update is also stored to be able to calculate future rewards from this point onwards.
+
+- When there is an incentive, the following variables get updated:
+  - The `rewardRate` (it depends directly on the amount of total rewards)
+  - The `rewardPerTokenStored` (rewardRate per vote before updating to new rewardRate)
+  - Time of the update (to current time of the incentive)
+  - The missing rewards if there are no allocations
+- In case of allocating, the following variables get updated:
+  - The `rewardPerTokenStored` (it depends on the amount of votes)
+  - Time of the update (to current time of the incentive)
```
Comment (antomor) [inline docs/RewardsCalculation.md:104]:
```suggestion
  - Time of the update (to current time of the allocation)
```

**Review by franciscotobar — APPROVED** [review]
LGTM

**Review by antomor — APPROVED** [review]

**Pull request merged**